packages/dds/merge-tree/src/mergeTreeNodeWalk.ts

```diff
@@ -178,7 +178,7 @@ export function walkAllChildSegments(
         startBlock.children[0],
         startBlock.parent === undefined
             ? undefined
-            : (node) => node === startBlock.parent ? NodeAction.Exit : NodeAction.Continue,
+            : (node) => node.parent === startBlock.parent ? NodeAction.Exit : NodeAction.Continue,
```

Comment:
won't this exit immediately without walking any nodes in the non-root case?

Reply:
did the previous code actually fail a test? as i still think it was correct. node won't equal start.parent util we've walked all of start, and then when up a layer above start, then we'd down action on that next layer and node would equal start.parent so we'd exit

Reply:
i actually think we there is a problematic case if start is the last child of its parent, as that will cause up to go up too far, and down won't see the parent again

Reply:
i'd considered allowing us to exit on up action but didn't think i needed it anywhere. we might need it here

Reply:
ahh i get it now :)

Reply:
> won't this exit immediately without walking any nodes in the non-root case?

no, as you've since realized :)

> i actually think we there is a problematic case if start is the last child of its parent, as that will cause up to go up too far, and down won't see the parent again

Yeah, you're totally right. Thanks for catching. I've updated the implementation to check ancestors. Using an up action might be better, but seems like a more involved refactor for not much gain.